src/mongocxx/write_concern.cpp

```diff
@@ -65,9 +65,20 @@ void write_concern::nodes(std::int32_t confirm_from) {
 }
 
 void write_concern::acknowledge_level(write_concern::level confirm_level) {
-    libmongoc::write_concern_set_w(
-        _impl->write_concern_t,
-        static_cast<std::underlying_type<write_concern::level>::type>(confirm_level));
+    std::int32_t w;
+    switch (confirm_level) {
+    case write_concern::level::k_unacknowledged:
```

Comment:
Since we're doing the translation between write_concern::level and integer values in a couple of places, maybe we should pull these out into constants somewhere